backend/app/api/routes/auth.py

```
@@ -1,226 +1,448 @@
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"OAuth initialization failed for provider {oauth_data.provider}"
+        )
+
+
+@router.get("/oauth/callback")
+async def oauth_callback(
+    code: str = Query(..., description="Authorization code from OAuth provider"),
+    state: str = Query(..., description="State parameter for CSRF protection"),
+    provider: str = Query(..., description="OAuth provider name"),
+    db: AsyncSession = Depends(get_db)
```

Ruff (0.13.1)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)